daemon/list.go

```diff
@@ -393,15 +397,24 @@ func (daemon *Daemon) transformContainer(container *container.Container, ctx *li
 // Volumes lists known volumes, using the filter to restrict the range
 // of volumes returned.
 func (daemon *Daemon) Volumes(filter string) ([]*types.Volume, error) {
-	var volumesOut []*types.Volume
+	var (
+		volumesOut []*types.Volume
+		filterUsed = false
+	)
 	volFilters, err := filters.FromParam(filter)
 	if err != nil {
 		return nil, err
 	}
-
-	filterUsed := volFilters.Include("dangling") &&
-		(volFilters.ExactMatch("dangling", "true") || volFilters.ExactMatch("dangling", "1"))
-
+	if err := volFilters.Validate(acceptedVolumeFilterTags); err != nil {
+		return nil, err
+	}
+	if volFilters.Include("dangling") {
+		if volFilters.ExactMatch("dangling", "true") {
```

Comment:
Hi @cpuguy83 if we checked docker images filter with dangling, it not allowed 1, only allow true.
$ docker images -f dangling=1
Error response from daemon: Invalid filter 'dangling=1'

So I think it is better to keep consistent, whether change image part, or not used "1" in volume